docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -494,51 +441,58 @@ class ComponentExample extends React.Component<ComponentExampleProps, ComponentE
   }
 
   private renderError = () => {
-    const { error } = this.state
-
-    if (!error) return null
-
     return (
-      <Segment size="small" color="red" basic inverted padded secondary>
-        <pre>{error}</pre>
-      </Segment>
+      <SourceRender.Consumer>
```

Comment:
Context allows to avoid the storage of render related stuff in the state of `ComponentExample` and avoid cycling updates like this: `render ComponentExample` => `render dource` =>`update state` => `render ComponentExample` (again).

Reply:
sorry, cannot completely understand how it is different with the context.

Lets, for the sake fo simplicity, consider the following case: we have _code pane_ and _render pane_, where code is mapped to rendered DOM.

Also, suppose that there is `ComponentExample` component who manages these two.

Initially code and rendered DOM are rendered at the corresponding panes. After that we start editing code:

### State-based approach

The following will happen if `ComponentExample`'s `state` will be used:

```
`RENDER PANE`
-----------------
`CODE PANE` -> content changed ->`code` evaluated to `state.renderedTree`, RERENDER
```

### Context-based approach

So, one rerendering for state (two if this process will be debounced) - no extra unexpected cycles. How it is different from context-based way? What we will have is the following (under assumption that debouncing is removed, as we have it now):

```
`RENDER PANE`
-----------------
`CODE PANE` -> content changed -> evaluated to `state.renderedTree`, RERENDER with updated context provider
```

So, still one update, but why do we need context indirection?

-----------

Please, help me settle these things in my head. Thank you!